datumaro/components/operations.py

```diff
@@ -163,45 +201,91 @@ def _merge_images(item_a: DatasetItem, item_b: DatasetItem) -> Image:
-                image = item_b.image
-            elif item_a.image.has_size:
-                image = item_a.image
-            elif item_b.image.has_size:
-                image = item_b.image
+            if item_a.media.has_data:
+                media = item_a.media
+            elif item_b.media.has_data:
+                media = item_b.media
+            elif item_a.media.path:
+                media = item_a.media
+            elif item_b.media.path:
+                media = item_b.media
+            elif item_a.media.has_size:
+                media = item_a.media
+            elif item_b.media.has_size:
+                media = item_b.media
             else:
                 assert False, "Unknown image field combination"
 
-            if not image.has_data or not image.has_size:
-                if item_a.image._size:
-                    image._size = item_a.image._size
-                elif item_b.image._size:
-                    image._size = item_b.image._size
-        elif item_a.has_image:
-            image = item_a.image
+            if not media.has_data or not media.has_size:
+                if item_a.media._size:
+                    media._size = item_a.media._size
+                elif item_b.media._size:
+                    media._size = item_b.media._size
+        elif isinstance(item_a.media, Image):
+            media = item_a.media
+        else:
+            media = item_b.media
+
+        return media
+
+    @staticmethod
+    def _merge_point_clouds(item_a: DatasetItem, item_b: DatasetItem) -> PointCloud:
+        media = None
+
+        if isinstance(item_a.media, PointCloud) and isinstance(item_b.media, PointCloud):
+            if item_a.media.path and item_b.media.path and \
+                    item_a.media.path != item_b.media.path:
+                raise MismatchingMediaPathError(
+                    (item_a.id, item_a.subset),
+                    item_a.media.path, item_b.media.path)
+
+            if item_a.media.path or item_a.media.extra_images:
+                media = item_a.media
+
+                if item_b.media.extra_images:
+                    for image in item_b.media.extra_images:
+                        if image not in media.extra_images:
+                            media.extra_images.append(image)
+            else:
+                media = item_b.media
+
+                if item_a.media.extra_images:
+                    for image in item_a.media.extra_images:
+                        if image not in media.extra_images:
+                            media.extra_images.append(image)
+
+        elif isinstance(item_a.media, PointCloud):
+            media = item_a.media
+        else:
+            media = item_b.media
+
+        return media
+
+    @staticmethod
+    def _merge_videos(item_a: DatasetItem, item_b: DatasetItem) -> Video:
+        media = None
+
+        if isinstance(item_a.media, Video) and isinstance(item_b.media, Video):
+            raise VideoMergeError(item_a.id)
```

Comment:
Does it mean we can never merge video datasets? I shouldn't be so. We can merge videos if they have the same paths and frame options.